### PR TITLE
[#97] Switch autoreleases from hub to gh

### LINKED ISSUE #97: Use the official GitHub CLI to create releases
# Clarification and motivation

We are using the `hub` utility to interact with GitHub from CI, but I heard that there is another tool which is officially developed by GitHub and lets you interact with GitHub. Probably it's quite similar to `hub`. I think we should consider switching to it because "being developed by GitHub" sounds like an essential advantage here. 

# Acceptance criteria

1. Evaluate what should be done in order to switch to this tool.
2. If the change is considered good enough, do the switch.

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: There is an official CLI from github, now it can handle
releases.

Solution: Switch to it from hub tool.

## Related issue(s)

Resolves #97 

#### Related changes (conditional)

- [x] I checked whether I should update the [README](../tree/master/README.md)

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).

### CODE REVIEW

File: scripts/autorelease.sh [hunk 1]
```diff
@@ -37,9 +37,11 @@ git push --force --tags
 # Combine all assets
 assets=()
 for file in $assets_dir/*; do
-    echo $file
-    assets+=("-a" $file)
+    assets+=($file)
 done
```
Comment (zhenyavinogradov) [inline scripts/autorelease.sh:41]: Looks like you can simply pass `$assets_dir/*` to `gh release upload` call below

File: scripts/shell.nix [hunk 2]
```diff
@@ -2,8 +2,28 @@
...
+          buildGoModule = args: super.buildGoModule (args // rec {
+            version = "1.2.0";
+            vendorSha256 = "0ybbwbw4vdsxdq4w75s1i0dqad844sfgs69b3vlscwfm6g3i9h51";
+            src = self.fetchFromGitHub {
+              owner = "cli";
+              repo = "cli";
+              rev = "v${version}";
+              sha256 = "17hbgi1jh4p07r4p5mr7w7p01i6zzr28mn5i4jaki7p0jwfqbvvi";
+            };
+          });
```
Comment (zhenyavinogradov) [inline scripts/shell.nix:19]: Just to note, in general it's better to push/cherry-pick updates to packages like this into our nixpkgs fork. A cherry-picked commit would even be automatically overlayed on our next nixpkgs rebase.

**Review by zhenyavinogradov — APPROVED** [review]

File: scripts/autorelease.sh (hunk 1 above)
Comment (rvem) [reply scripts/autorelease.sh:41]: :smart:

File: scripts/autorelease.sh [hunk 3]
```diff
@@ -27,19 +27,15 @@ shopt -s extglob
...
-    echo $file
-    assets+=("-a" $file)
-done
-
 # Create release
-hub release create "${assets[@]}" -F "$TEMPDIR"/"$project"/release-notes.md --prerelease auto-release
+gh release create -F "$TEMPDIR"/"$project"/release-notes.md --prerelease auto-release
+
+# Upload assets
+gh release upload auto-release "$assets_dir/*"
```
Comment (zhenyavinogradov) [inline scripts/autorelease.sh:41]: `*` won't expand inside quotes :pk:

**Pull request merged**